rfcs/0189-batch-task-apis.md

```diff
@@ -0,0 +1,126 @@
+# RFC 189 - Batch APIs for task definition, status and index path
+* Comments: [#189](https://github.com/taskcluster/taskcluster-rfcs/pull/189)
+* Proposed by: @Alphare and @ahal
+
+# Summary
+
+Add API endpoints to query the definition, status or index paths of multiple
+tasks in a single call.
+
+## Motivation
+
+When looking at Decision task profiles in Gecko, it was noticed that nearly 70%
+of the runtime (representing ~3 minutes) was spent waiting on queries to two
+Taskcluster APIs:
+
+1. `/task/<taskId>/status`
+2. `/task/<indexPath>`
+
+Each individual call is fairly quick, but Gecko Taskgraph's optimization phase
+can make thousands of these requests. Creating an API that can return all the
+information Taskgraph needs in a handful of API requests would greatly speed up
+the overall time the Queue and Index services spend looking things up in the
+database, as well as the time Gecko Decision tasks spend waiting on the
+network.
+
+Note: Taskgraph doesn't actually use the `task/<taskId>` endpoint here, but
+this endpoint is adjacent to the other two, so for consistency it may make
+sense to implement a batch API for that as well.
+
+### Proof of Concept
+
+A proof of concept was created whereby the requests to Taskcluster were simulated
+such that all data could be obtained in a single API call. The overal Decision task
+time was reduced by ~3 minutes.
```

Comment:
oh wow. that's huge!

Reply:
It looks like I obtained this figure with a setup that wasn't representative of the actual real-world use of the profile command. The savings are still significant, about 20% of the entire runtime. Not 75%, alas. :)